modules/renter/repair.go

```diff
@@ -26,323 +25,374 @@ var renewThreshold = func() types.BlockHeight {
 	case "dev":
 		return 200
 	default:
-		return 144 * 7 * 3 // 3 weeks - to soon be 6 weeks.
+		return 144 * 7 * 6
 	}
 }()
 
-// hostErr and hostErrs are helpers for reporting repair errors. The actual
-// Error implementations aren't that important; we just need to be able to
-// extract the NetAddress of the failed host.
-
-type hostErr struct {
-	host modules.NetAddress
-	err  error
-}
-
-func (he hostErr) Error() string {
-	return fmt.Sprintf("host %v failed: %v", he.host, he.err)
+type chunkGaps struct {
+	contracts []types.FileContractID
+	pieces    []uint64
+	numGaps   int
 }
 
-type hostErrs []*hostErr
-
-func (hs hostErrs) Error() string {
-	var errs []error
-	for _, h := range hs {
-		errs = append(errs, h)
+func (r *Renter) addFileToRepairMatrix(file *file, availableWorkers map[types.FileContractID]struct{}, repairMatrix map[string]*chunkGaps, gapCounts map[int]int) {
+	// Flatten availableWorkers into a list of contracts.
+	contracts := make([]types.FileContractID, 0)
+	for contract := range availableWorkers {
+		contracts = append(contracts, contract)
 	}
-	return build.JoinErrors(errs, "\n").Error()
-}
 
-// repair attempts to repair a file chunk by uploading its pieces to more
-// hosts.
-func (f *file) repair(chunkIndex uint64, missingPieces []uint64, r io.ReaderAt, hosts []contractor.Editor) error {
-	// read chunk data and encode
-	chunk := make([]byte, f.chunkSize())
-	_, err := r.ReadAt(chunk, int64(chunkIndex*f.chunkSize()))
-	if err != nil && err != io.EOF && err != io.ErrUnexpectedEOF {
-		return err
+	// If the file is not being tracked, don't add it to the repair matrix.
+	_, exists := r.tracking[file.name]
+	if exists {
+		return
```

Comment:
shouldn't this be in `createRepairMatrix`? Then `addFileToRepairMatrix` doesn't need to be a method of the Renter.